Blue & Red (v1) support, IMU with attitude mode

samuelpowell opened the pull request.

bartslinger committed:
blue board xn297 first flight achieved

samuelpowell committed:
Merge pull request #1 from bartslinger/xn297_blue_board

Merge blue board support

samuelpowell committed:
Remove builds and unused configuration files

samuelpowell committed:
Build with Keil

samuelpowell committed:
Modify bind code to respond with supplied TXID

Successfully binds with stock TX

samuelpowell committed:
Wait for reply containing submitted AID during bind

samuelpowell committed:
Implement channel hopping

CX10 receiving data, can arm and run motors

samuelpowell committed:
Merge pull request #3 from samuelpowell/bluebart

Support blue PCB CX-10

samuelpowell committed:
Integrate nrf24 and xn297 radio drivers

samuelpowell committed:
Rename files according to integrate RF driver and protocols, remove PPM

samuelpowell committed:
Update licensing information for brevity

samuelpowell committed:
Clean up nRF driver

samuelpowell committed:
Clean up blue protocol

samuelpowell committed:
Refactor RF code

samuelpowell committed:
Rename failsave

samuelpowell committed:
Align blue operation with red

samuelpowell committed:
Re-enable peripherals with new define structure

bartslinger commented:
Good to see you working on this! Another interesting project is this one: https://github.com/goebish/nrf24_multipro.
That also works with the blue board.

samuelpowell committed:
Move LED operation to timer driven interrupt

samuelpowell commented:
@bartslinger, all the recent work getting the XN297 to play nicely with the nrf24 motivated me play around a little more. I have merged your fork and recognised your contribution in the copyright statements and license document, I assume that's okay?

bartslinger commented:
@samuelpowell Yep that's definitely okay :-)

samuelpowell committed:
Refactor timers and motor control

samuelpowell committed:
Main code cleanup pending PID extraction

samuelpowell committed:
Remove serial comms

Newer boards do not have available ports and thus this limits to debug,
which can be done via SWD anyway.

samuelpowell committed:
SystemInit is called by startup code

samuelpowell committed:
Remove further serial related functionality, update redv1 build

samuelpowell commented:
@bartslinger, do you know what is powered by the LDO on the BLUE PCB?

samuelpowell committed:
More constants

samuelpowell committed:
Implement state machine

samuelpowell committed:
Remove IMU RX and I2C globals

bartslinger commented:
@samuelpowell I think at least the STM32, but probably also the radio, IMU and LEDs?

samuelpowell committed:
Refactor main state machine

samuelpowell commented:
@bartslinger, ah, okay: so when you turn it off in your code, there's no coming back?

samuelpowell committed:
Cleanup protocols

samuelpowell committed:
Combine protocols

bartslinger commented:
@samuelpowell No, I believe there is a voltage check somewhere that turns it off if the voltage is too low. When the switch is put to ON, I think there is a capacitor or something that provides power for half a second. In this time, the LDO should be turned on or the CX-10 will shutdown immediately.

By the way, you might be interested in this: http://www.bartslinger.com/category/cx-10-quadcopter/
Although I think you figured most of that out already.

samuelpowell committed:
Integrate protocols save for binding

samuelpowell committed:
Merge pull request #5 from samuelpowell/protocombine

Combine RF protocols (part 1)

samuelpowell commented:
@bartslinger, thanks, that makes sense. The blog is great, well done on the award!

samuelpowell committed:
Clean up

samuelpowell committed:
Merge pull request #6 from samuelpowell/imuinv

Clean up

samuelpowell committed:
Revert "Clean up"

samuelpowell committed:
Merge pull request #7 from samuelpowell/revert-6-imuinv

Revert "Clean up"

samuelpowell committed:
Clean up again

samuelpowell committed:
Merge pull request #8 from samuelpowell/imuinv

Clean up again

samuelpowell committed:
Naive (non functional) quaternion IMU

samuelpowell committed:
Flight tested attitude mode.

Works to some degree, gains require tuning.

samuelpowell committed:
Merge pull request #9 from samuelpowell/imu1

Implement IMU and attitude mode

samuelpowell committed:
Fix arm/disarm

Bug in orientation:

IMU sees gyro rpy inputs as p-ry, although accelerometers are correct.

samuelpowell committed:
IMU gyro + acc consistent

samuelpowell committed:
Modest acc gain

samuelpowell committed:
Angles and directions are now consistent on BLUE

samuelpowell committed:
Refactor control, invert elevator in RX to match IMU

samuelpowell committed:
Use calibrated units in all flight modes

PID still operates in integers through prescaling.

Removed more globals.

samuelpowell committed:
Merge pull request #10 from samuelpowell/imu4

Use calibrated units in all flight modes

samuelpowell committed:
Merge pull request #11 from samuelpowell/imu2

Fix alignment, use calibrated units

samuelpowell committed:
Switch flight modes on AUX2

samuelpowell committed:
Update mixer with new orientations

samuelpowell committed:
Flight test

Attitude P gain too high, causes oscillations. Accelerometer error gain
currently zero, attitude drifts. Reduce P gain until no oscilation,
then add in accelerometer error gain.
